From 9231f1bdaac2f2b5ebc7135e5ee87f716e944f46 Mon Sep 17 00:00:00 2001
From: Yang Fan F <fan.f.yang@intel.com>
Date: Fri, 5 Feb 2016 19:56:24 +0800
Subject: [PATCH 02/24] atomisp:Fix klockwork critical issue in atomisp driver

in atomisp_v4l2.c: the void __iomem *base is not checked if it's null.

in hmm_bo.c: the struct scatterlist *sl is not checked if it's null.

Change-Id: I065a52da4d5ada347336ee52702dcf9516f52fed
Original-Tracked-On: #H1504149861
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/97207
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Reviewed-by: DaleiX Liu <daleix.liu@intel.com>
Reviewed-by: GuoqingX Zhang <guoqingx.zhang@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-18671
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/466974
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c   |  5 +++++
 .../drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c     | 10 +++++++++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index 93615df..a8cdb3e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -1300,6 +1300,11 @@ static int atomisp_pci_probe(struct pci_dev *dev,
 	}
 
 	base = pcim_iomap_table(dev)[ATOM_ISP_PCI_BAR];
+	if (!base) {
+		dev_err(&dev->dev, "atomisp: error iomap table ptr\n");
+		return -EINVAL;
+	}
+
 	dev_dbg(&dev->dev, "base: %p\n", base);
 
 	atomisp_io_base = base;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
index 196c554..a44d939 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
@@ -1042,7 +1042,15 @@ static int alloc_ion_pages(struct hmm_buffer_object *bo,
 
 	sl = sg_tbl->sgl;
 	for_each_sg(sg_tbl->sgl, sl, sg_tbl->nents, i) {
-		struct page *page = sg_page(sl);
+		struct page *page;
+
+		if (!sl) {
+			dev_err(atomisp_dev, "sg is NULL.\n");
+			ret = -EINVAL;
+			goto error;
+		}
+
+		page = sg_page(sl);		
 		for (j = 0; j < sl->length / PAGE_SIZE; j++) {
 			bo->page_obj[page_nr++].page = page++;
 			/*
-- 
1.9.1

